### PR TITLE
Make Hashicorp setup apply to HCP/ Enterprise

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1642

### CODE REVIEW

File: astro/secrets-backend.md [hunk 1]
```diff
@@ -169,26 +169,38 @@ If you do not already have a Vault server deployed but would like to test this f
 
 To use Vault as a secrets backend, Astronomer recommends configuring a Vault AppRole with a policy that grants only the minimum necessary permissions for Astro. To do this:
 
-1. [Create a Vault policy](https://www.vaultproject.io/docs/concepts/policies) with the following permissions:
+1. Run the following command to [create a Vault policy](https://www.vaultproject.io/docs/concepts/policies) that Astro can use to access Vault:
```
Comment (kwcanuck) [inline astro/secrets-backend.md:172]:
```suggestion
1. Run the following command to [create a Vault policy](https://www.vaultproject.io/docs/concepts/policies) that Astro can use to access a Vault server:
```

File: astro/secrets-backend.md [hunk 2]
```diff
@@ -169,26 +169,38 @@ If you do not already have a Vault server deployed but would like to test this f
...
+        token_num_uses=0 \
+        token_ttl=24h \
+        token_max_ttl=24h
     ```
 
-2. [Create a Vault AppRole](https://www.vaultproject.io/docs/auth/approle) and attach the policy you just created to it.
-   
-3. Retrieve the `role-id` and `secret-id` for your AppRole by running the following commands:
+3. Add the policy you created to your AppRole. See [Associate Policies to Auth Methods](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-policies#associate-policies-to-auth-methods).     
+4. Retrieve the `role-id` and `secret-id` for your AppRole by running the following commands:
```
Comment (kwcanuck) [inline astro/secrets-backend.md:196]:
```suggestion
4. Run the following commands to retrieve the `role-id` and `secret-id` for your AppRole:
```

File: astro/secrets-backend.md [hunk 3]
```diff
@@ -169,26 +169,38 @@ If you do not already have a Vault server deployed but would like to test this f
...
 
     ```sh
     vault read auth/approle/role/<your-approle>/role-id
     vault write -f auth/approle/role/<your-approle>/secret-id
     ```
+
+    Save these values for later in the setup.
```
Comment (kwcanuck) [inline astro/secrets-backend.md:203]:
```suggestion
    Save these values. You'll need them later to complete the setup.
```

**Review by kwcanuck — APPROVED** [review]

**Comment by kwcanuck** [comment]
@jwitz what remains to get this PR merged?

**Comment by jwitz** [comment]
@kwcanuck I'm waiting on @jacobm3 's review, will ping again!

**Review by jacobm3 — CHANGES_REQUESTED** [review]
The additional info block with `"namespace": "admin"` looks great and is the precise info needed for HCP use, and enough of a pointer for Vault Enterprise use. 

I would not add the `vault secrets disable secret` commands. They are not needed if the reader is starting with a clean Vault installation and they could cause data loss if run on a Vault server that already has secrets at that mount point.

File: astro/secrets-backend.md [hunk 4]
```diff
@@ -197,12 +209,16 @@ To start, create an Airflow variable or connection in Vault that you want to sto
 To store an Airflow variable in Vault as a secret, run the following Vault CLI command with your own values:
 
 ```sh
+vault secrets disable secret
```
Comment (jwitz) [inline astro/secrets-backend.md:212]:
```suggestion
```

File: astro/secrets-backend.md [hunk 5]
```diff
@@ -197,12 +209,16 @@ To start, create an Airflow variable or connection in Vault that you want to sto
...
+vault secrets enable -path=secret -version=2 kv
 vault kv put secret/variables/<your-variable-key> value=<your-variable-value>
 ```
 
 To store a connection in Vault as a secret, run the following Vault CLI command with your own values:
 
 ```sh
+vault secrets disable secret
```
Comment (jwitz) [inline astro/secrets-backend.md:220]:
```suggestion
```

**Comment by jwitz** [comment]
Gonna go ahead and merge this because the requested changes have been addressed.

**Pull request merged**